lib/src/client.dart

```diff
@@ -955,13 +957,64 @@ class Client extends MatrixApi {
   final Map<String, ProfileInformation> _profileRoomsCache = {};
   final Map<String, ProfileInformation> _profileServerCache = {};
 
+  /// Get the combined profile information for this user. First checks for a
+  /// non outdated cached profile before requesting from the server. Cached
+  /// profiles are outdated if they have been cached in a time older than the
+  /// [maxCacheAge] or they have been marked as outdated by an event in the
+  /// sync loop.
+  /// In case of an
+  ///
+  /// [userId] The user whose profile information to get.
+  @override
+  Future<CachedProfileInformation> getUserProfile(
+    String userId, {
+    Duration timeout = const Duration(seconds: 30),
+    Duration maxCacheAge = const Duration(days: 3),
+  }) async {
+    final cachedProfile = await database?.getUserProfile(userId);
+    if (cachedProfile != null &&
+        (cachedProfile.outdated != true ||
+            DateTime.now().difference(cachedProfile.updated) >= maxCacheAge)) {
+      return cachedProfile;
+    }
+
+    late final Future<ProfileInformation> request;
+    try {
+      request = _userProfileRequests[userId] ??=
+          super.getUserProfile(userId).timeout(timeout);
+    } catch (e) {
+      Logs().d('Unable to fetch profile from server', e);
+      if (cachedProfile == null) rethrow;
+      return cachedProfile;
+    }
+
+    final profile = await request;
```

Comment:
if I'm not wrong, the exception will occur here only and not in the `request` assignment.

Reply:
You are right I think 🤔 thank you so much. Fixing it